src/app/pages/task/task.page.ts

```diff
@@ -80,25 +93,93 @@ export class TaskPage implements OnInit {
   }
 
   private async loadData() {
-    // Refresh cache first
-    await this.itemService.refreshItems().catch(() => {
-      this.presentToast('Cannot refresh items from server');
-    });
     // Subscribe to local cache changes
-    this.itemService.getItems().subscribe(result => {
+
+    // kinda hacky but before calling subscribe
+    // we get the last result
+    // if loading is true and fetch policy is 'cache-only'
+    // then we need to do a resync
+    // Again this behaviour (or similar) should be done by offix
+    const result = await this.getTasksQuery.result()
+    this.lastSync = result.data.allTasks.lastSync
+    if (result.loading && this.getTasksQuery.options.fetchPolicy === 'cache-only') {
+      await this.sync()
+    }
+    
+
+    this.getTasksQuery.subscribe(result => {
       if (result && !result.errors) {
         console.log('Result from query', result);
-        this.items = result.data && result.data.allTasks;
+        this.lastSync = result.data.allTasks.lastSync
+        this.items = result.data && result.data.allTasks.items;
       } else {
         console.log('error from query', result);
-        this.presentToast('Cannot load data from cache');
+        this.presentToast('Cannot load data');
       }
     }, error => {
       console.log('error from query', error);
       this.presentToast('Problem with listening to cache changes.');
     });
   }
 
+  // This should probably be a helper in offix
+  // ideally we'll be able do define a query like
+  
+  // const getTasks = buildSyncQuery({ query: GET_TASKS, variables: {} ...})
+  // and then manually call getTasks.sync() or that it would be network aware and do things by itself.
+  sync() {
+    const lastSync = this.lastSync;
+    const queryName = this.getTasksQuery.queryName
+    
+    // the first time we perform the query it might have been against the cache
+    // Then we set it back to 'cache-and-network' when we fetch additional data
+    // a little hacky but can be abstracted away in offix
+    this.getTasksQuery.options.fetchPolicy = 'cache-and-network'
+
+    // fetchmore lets you call the same query again with different params
+    // and have the results merged back under the same query within the cache
+    this.getTasksQuery.fetchMore({
```

Comment:
Ping @eunovo - Most important method that executes diff query.
Then it manually does cache updates using updateQuery method